e2e-tests/src/NPG/helper.js

```diff
@@ -141,34 +144,79 @@ export const retrievePaymentRedirectUrl = async (walletHost, walletToken) => {
+                details: {
+                  detailType: "wallet",
+                  walletId: walletId
+                }
               }),
             })
+            console.debug('post auth request');
             if (responseAuthRequest.status === 200) {
-
-              console.debug('auth request 200');
+              console.debug('post auth request 200');
               const authRequest = await responseAuthRequest.json();
               return authRequest.authorizationUrl;
             } else {
-              throw Error('Error during auth request');
+              throw Error(`Error during auth request: status code ${responseAuthRequest.status}`);
             }
           } else {
-            throw Error('Error getting fees');
+            throw Error(`Error getting fees: status code ${responseFees.status}`);
           }
         } else {
-          throw Error('Error during transaction');
+          throw Error(`Error during transaction: status code ${responsePaymentTransaction.status}`);
+        }
+        } else {
+          throw Error(`Error getting wallet: status code ${responseUserWallet.status}`);
         }
       } else {
-        throw Error('Error getting rptId');
+        throw Error(`Error getting rptId: status code ${responseRptIdInfo.status}`);
       }
     } else {
-      throw Error('Error starting session');
+      throw Error(`Error starting session: status code ${responseStartSession.status}`);
     }
-  } else {
-    throw Error('Error getting wallet');
-  }
 };
 
+export const clickPaypalButton = async () => {
+  const preambleButton = '#preambleButton'
+  await page.click(preambleButton)
+}
+
+export const checkAndClickPaypalFirstPsps = async () => {
+  const radioButtonIntesa = await page.waitForSelector('#BCITITMM')
+  await radioButtonIntesa.click()
+  const submitPsp = await page.waitForSelector('#apmSubmit')
+  await submitPsp.click()
+}
+
+export const fillPaypalAuth = async paypalCredentials => {
+  const usernameInput = '#email';
+  const btnNext = '#btnNext';
+  const pwInput = '#password';
+  const loginButton = '#btnLogin';
+  const payButton = '#consentButton';
+
+  const targets = await browser.targets()
+  const popup = await targets[targets.length -1].page()
+  await popup.waitForNavigation()
+  await popup.waitForSelector(usernameInput, {timeout: 10000});
+  await popup.click(usernameInput, { clickCount: 3 });
+  await popup.keyboard.type(paypalCredentials.username);
+  await popup.waitForSelector(btnNext, {timeout: 10000});
+  await popup.click(btnNext);
+  await popup.waitForSelector(pwInput, {timeout: 10000});
+  await new Promise(r => setTimeout(r, 2000));
+  await popup.focus(pwInput);
+  await new Promise(r => setTimeout(r, 2000));
+  popup.keyboard.type(paypalCredentials.password);
+  await new Promise(r => setTimeout(r, 2000)); 
+  const login = await popup.waitForSelector(loginButton, {timeout: 10000, visible: true});
+  await new Promise(r => setTimeout(r, 2000)); 
```

Comment:
Isn't one delay sufficient?

Reply:
I added this timeout because paypal view does some strange transitions between the steps. And also when a selector is detected it doesn't seem to be ready to receive input. With this timeout it seems to be more reactive to puppeteer input.